src/module_2/module_2_1.md

```diff
@@ -0,0 +1,402 @@
+```python
+import numpy as np
+import pandas as pd
+import os
+
+
+folder_path = "../../data" #Folder in main directory
+
+# # Crea el directorio local si no existe
+# if not os.path.exists(data_folder):
+#     os.makedirs(data_folder)
+```
+
+
+```python
+# Clean Users dataframe
+# Importing dataframe from "dirty" folder
+df = pd.read_parquet(f"{folder_path}/users.parquet")
+
+# Drop information about users with no adults
+drop_indexes = df[
+    ((df["count_adults"] == 0) & (df["count_people"] > 0))
+    | ((df["count_adults"] == 0) & (df["count_pets"] > 0))
+].index
+
+df_clean = df.copy()
+df_clean.drop(index=drop_indexes, inplace=True)
+
+df_clean["missing_values"] = np.where((df_clean["count_adults"].isna() | 
+                                      df_clean["count_children"].isna() |
+                                      df_clean["count_babies"].isna() |
+                                      df_clean["count_pets"].isna()),
+                                      1, 0)
+
+missing_columns = [
+    "count_adults",
+    "count_children",
+    "count_babies",
+    "count_pets",
+]
+
+for column in missing_columns:
+    df_clean[column].fillna(df_clean[column].mean(), inplace=True)
```

Comment:
What's the potential issue of fillna with mean in variables with some many missing values? We will end up with `delta-like` distributions where most of the values are the same and equal to that original mean.